easybuild/easyblocks/m/mesa.py

```diff
@@ -0,0 +1,17 @@
+# This easyblock was created by the BEAR Software team at the University of Birmingham
+
+from easybuild.easyblocks.generic.mesonninja import MesonNinja
+from easybuild.tools.systemtools import POWER, X86_64, get_cpu_architecture
+
+
+class EB_Mesa(MesonNinja):
+    def configure_step(self, cmd_prefix=''):
+        """
+        Customise the configopts based on the platform
+        """
+        arch = get_cpu_architecture()
+        if arch == X86_64:
+            self.cfg.update('configopts', "-Dgallium-drivers='swrast,swr' -Dswr-arches=avx,avx2,skx,knl")
```

Comment:
@edmondac Can you clarify what this does exactly? Does this yield a "fat" build of Mesa, which works on different processor architectures?
Any links to Mesa documentation for this?

Reply:
This is what was in https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/m/Mesa/Mesa-19.2.1-GCCcore-8.3.0.eb

I can revert to the https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/m/Mesa/Mesa-19.1.7-GCCcore-8.3.0.eb options if you like, which doesn't have the `-Dswr-arches` flag at all...

Reply:
And I don't know what the difference is... I just didn't want to lose a potential improvement. The swr-arches bit in the later easyconfig is all about x86 so wouldn't work on power9, for example.

Reply:
Maybe @wpoely86 is familiar with this?

Reply:
Or @bartoldeman, since the `Mesa` easyconfig that includes this was contributed by him in https://github.com/easybuilders/easybuild-easyconfigs/pull/9143

Reply:
As far as I remember, swr by default builds a version for avx, for avx2, etc and picks the correct one at runtime.

Reply:
I now think that it shouldn't have this bit at all, since it won't build on anything older than Skylake...
```
-Dswr-arches=avx,avx2,skx,knl"
```
Patch coming.

A specific easyconfig could still add that in if desired, like the one in easybuilders/easybuild-easyconfigs#9143